### ISSUE TITLE
addParent to RootNode ?

### ISSUE THREAD

**Issue by mritunjaymusale** [KristianAbrahamsen/tree-util#4]
Can you provide an example of how to add a parent to existing root node?

**Comment by KristianAbrahamsen** [comment]
Hello mritunjaymusale

That is not yet supported, but I think it an interesting feature.
The new parent will then of course be the new root node of the tree.

**Comment by mritunjaymusale** [comment]
Yes, this will make it a lot more easier for use cases like mine where data fetched from api can be automatically set as the parent of the root node before processing

**Comment by mritunjaymusale** [comment]
subsequently I wanted to know is it possible to add multiple parents to single node?

**Comment by Zach-Jaensch** [comment]
@KristianAbrahamsen I am interesting in working on the add parent to root node feature (Not sure how I will go), but I don't think having multiple parents is a good idea as that is not a tree, but a graph data type https://www.javatpoint.com/tree-vs-graph-data-structure.

**Comment by KristianAbrahamsen** [comment]
@mritunjaymusale and @Zach-Jaensch

I do not intend to allow multiple parent nodes since it will not longer be a tree structure.
I might in the future allow to add one node to the parent, which in that case will be the new parent node.

**Comment by Zach-Jaensch** [comment]
@KristianAbrahamsen Would you accept a PR if I can find a solution?

**Comment by KristianAbrahamsen** [comment]
@Zach-Jaensch Sorry for late reply. Yes I will de delighted to accept a PR if you find a solution. Make sure you add some tests :)
